sdk/iOS/test/MSSyncTableTests.m

```diff
@@ -1082,9 +1083,348 @@ -(void) testPushNetworkTimeoutWithErroredOps
     XCTAssertEqual(filteredClient.syncContext.pendingOperationsCount, 3);
 }
 
+// Tests cancellation of the push operation at various stages of a push workflow.
+// - Verifies that a push following a cancelled push operation works as expected
+// - Verifies that cancelling a push operation finishes the push NSOperation regardless of the cancellation point
+// - Verifies that the expected number of items are pushed to the server
+-(void) testPushCancellability
```

Comment:
Ok, I'll make the assert message more detailed.
I'm avoiding splitting the 4 calls in into methods of their own because the only difference in them would be the cancellation points and they are kinda arbitrary points in the execution workflow. (Same for pull).